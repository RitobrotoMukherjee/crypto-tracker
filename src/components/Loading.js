const Loading = () => (
  <div className="flex flex-col items-center w-full h-screen">
    <h2 className="text-2xl font-bold">Loading...</h2>
  </div>
);

export default Loading;
