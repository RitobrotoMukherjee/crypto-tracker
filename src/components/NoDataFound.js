import PropTypes from 'prop-types';

const NoDataFound = ({ text }) => (
  <div className="flex flex-col items-center w-full h-screen">
    <h2 className="text-2xl font-bold">No Data Found</h2>
    <p className="text-lg font-title font-semibold">{text}</p>
  </div>
);

NoDataFound.propTypes = {
  text: PropTypes.string.isRequired,
};

export default NoDataFound;
