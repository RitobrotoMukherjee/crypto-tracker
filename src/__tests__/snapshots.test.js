import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/ConfigureStore';
import HomePage from '../components/Home';
import Loading from '../components/Loading';
import NoDataFound from '../components/NoDataFound';

describe('Main components snapshot testing', () => {
  it('Test if The Home Page renders correctly', () => {
    const TREE = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <HomePage />
        </BrowserRouter>
      </Provider>,
    );
    expect(TREE).toMatchSnapshot();
  });

  it('Test if The Home Page renders correctly', () => {
    const TREE = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Loading />
        </BrowserRouter>
      </Provider>,
    );
    expect(TREE).toMatchSnapshot();
  });

  it('Test if The Home Page renders correctly', () => {
    const TREE = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <NoDataFound text="Test Snapshot No Data Found" />
        </BrowserRouter>
      </Provider>,
    );
    expect(TREE).toMatchSnapshot();
  });
});
