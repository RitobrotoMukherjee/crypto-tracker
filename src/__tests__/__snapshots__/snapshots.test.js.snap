// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main components snapshot testing Test if The Home Page renders correctly 1`] = `
<div
  id="Home-holder"
>
  <div
    className="banner-container flex-col"
    id="search-bar"
  >
    <div
      className="overflow-hidden rounded my-3"
    >
      <img
        alt="Crypto Stats"
        className="object-cover w-32 h-32"
        src="logo.png"
      />
    </div>
    <div
      className="search-container"
    >
      <input
        className="w-full md:w-3/4 rounded p-3 text-gray-600"
        onChange={[Function]}
        placeholder="Search Coin"
        type="text"
        value=""
      />
      <button
        className="search-button"
        onClick={[Function]}
        type="button"
      >
        Clear Search
      </button>
    </div>
  </div>
  <div
    className="h-full"
    id="List"
  >
    <h1
      className="font-title uppercase pl-5 bg-titleBar"
    >
      Stats by coins
    </h1>
    <div
      className="flex flex-col items-center w-full h-screen"
    >
      <h2
        className="text-2xl font-bold"
      >
        Loading...
      </h2>
    </div>
    <div
      className="coin-grid"
      id="coin-grid"
    />
  </div>
</div>
`;

exports[`Main components snapshot testing Test if The Home Page renders correctly 2`] = `
<div
  className="flex flex-col items-center w-full h-screen"
>
  <h2
    className="text-2xl font-bold"
  >
    Loading...
  </h2>
</div>
`;

exports[`Main components snapshot testing Test if The Home Page renders correctly 3`] = `
<div
  className="flex flex-col items-center w-full h-screen"
>
  <h2
    className="text-2xl font-bold"
  >
    No Data Found
  </h2>
  <p
    className="text-lg font-title font-semibold"
  >
    Test Snapshot No Data Found
  </p>
</div>
`;
